core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -192,13 +191,11 @@ public List<DeleteFile> deletes() {
     @Override
     public CloseableIterable<StructLike> rows() {
       try (CloseableIterable<ManifestFile> manifests =
-          Avro.read(io.newInputFile(manifestListLocation))
-              .rename("manifest_file", GenericManifestFile.class.getName())
-              .rename("partitions", GenericPartitionFieldSummary.class.getName())
-              .rename("r508", GenericPartitionFieldSummary.class.getName())
+          InternalData.read(FileFormat.AVRO, io.newInputFile(manifestListLocation))
+              .setRootType(GenericManifestFile.class)
+              .setCustomType(508, GenericPartitionFieldSummary.class)
               .project(ManifestFile.schema())
               .classLoader(GenericManifestFile.class.getClassLoader())
-              .reuseContainers(false)
```

Comment:
This is just using the default, right?

Reply:
Yes, the new interface just has `reuseContainers()` to enable reuse, but no flag to change the behavior.